// @ringzero/nuxt3-front-template-set/dist/runtime/plugins/vuetify.mjs
import { createVuetify } from "vuetify";
import { VBtn } from "vuetify/components";
import { MAIN_THEME, mainTheme, mainDarkTheme } from "../helpers/themes.mjs";
import * as components from "vuetify/components";
import * as directives from "vuetify/directives";
import "@mdi/font/css/materialdesignicons.css";
import { defineNuxtPlugin } from "nuxt/app";
export default defineNuxtPlugin((nuxtApp) => {
  const vuetify = createVuetify({
    ssr: true,
    components,
    directives,
    display: {
      mobileBreakpoint: "sm"
    },
    // add theme
    theme: {
      defaultTheme: MAIN_THEME,
      themes: {
        mainTheme,
        mainDarkTheme
      },
      // primary-darken-9 primary-lighten-9 までできるようにする
      variations: {
        colors: ["primary", "secondary", "accent", "test", "tenant", "admin"],
        lighten: 9,
        darken: 9
      }
    },
    aliases: {
      MyButton: VBtn
    },
    defaults: {
      MyButton: {
        variant: "flat",
        height: 28,
        rounded: "lg",
        size: "small"
      }
    }
  });
  nuxtApp.vueApp.use(vuetify);
});
