// @ringzero/nuxt3-front-template-set/dist/runtime/components/common.mjs
export var LISTBUTTON_TYPE = /* @__PURE__ */ ((LISTBUTTON_TYPE2) => {
  LISTBUTTON_TYPE2["DISABLED"] = "disabled";
  LISTBUTTON_TYPE2["LINK"] = "simple";
  LISTBUTTON_TYPE2["DEFAULT"] = "default";
  return LISTBUTTON_TYPE2;
})(LISTBUTTON_TYPE || {});
export var FORM_MODE = /* @__PURE__ */ ((FORM_MODE2) => {
  FORM_MODE2["INLINE"] = "inline";
  FORM_MODE2["BOTH"] = "both";
  FORM_MODE2["DEFAULT"] = "default";
  FORM_MODE2["SIMPLE"] = "simple";
  return FORM_MODE2;
})(FORM_MODE || {});
export var FORM_ITEM = /* @__PURE__ */ ((FORM_ITEM2) => {
  FORM_ITEM2["INPUT"] = "input";
  FORM_ITEM2["PASSWORD"] = "password";
  FORM_ITEM2["TEXTAREA"] = "text";
  FORM_ITEM2["NUMERIC"] = "numeric";
  FORM_ITEM2["EMAIL"] = "email";
  FORM_ITEM2["NAME"] = "name";
  FORM_ITEM2["KANA"] = "kana";
  FORM_ITEM2["TEL"] = "tel";
  FORM_ITEM2["ZIPCODE"] = "zipcode";
  FORM_ITEM2["DATE"] = "date";
  FORM_ITEM2["DATETIME"] = "datetime";
  FORM_ITEM2["TIME"] = "time";
  FORM_ITEM2["SELECTTIME"] = "selecttime";
  FORM_ITEM2["MONTH"] = "month";
  FORM_ITEM2["CHECK_BOX"] = "check_box";
  FORM_ITEM2["RADIO_GROUP"] = "radio_group";
  FORM_ITEM2["CHECK_GROUP"] = "check_group";
  FORM_ITEM2["SELECT_MENU"] = "select_menu";
  FORM_ITEM2["SELECT_CHECK"] = "select_check";
  FORM_ITEM2["SUGGEST_SELECT_MENU"] = "suggest_select_menu";
  FORM_ITEM2["ASSOCIATE_SELECT_MENU"] = "associate_select_menu";
  FORM_ITEM2["ASSOCIATE_SELECT_CHECK"] = "associate_select_check";
  FORM_ITEM2["INVOICE"] = "invoice";
  FORM_ITEM2["LIST"] = "list";
  FORM_ITEM2["ICON_OR_LINK"] = "iconOrlink";
  FORM_ITEM2["FILE"] = "file";
  FORM_ITEM2["MULTI_FILE"] = "multi_file";
  FORM_ITEM2["HEADLINE"] = "headline";
  FORM_ITEM2["BLANK_COLUMN"] = "key_blank_column";
  FORM_ITEM2["ACTION"] = "key_action";
  FORM_ITEM2["DELETE"] = "key_delete";
  FORM_ITEM2["BLANK"] = "key_blank";
  FORM_ITEM2["LINE"] = "key_line";
  FORM_ITEM2["COMMENT"] = "key_comment";
  FORM_ITEM2["NOTE"] = "key_note";
  FORM_ITEM2["ATTENTION"] = "key_attention";
  FORM_ITEM2["WARNING"] = "key_warning";
  FORM_ITEM2["MODAL_LINK"] = "key_modallink";
  FORM_ITEM2["TERMS_CHECK"] = "key_termsCheck";
  FORM_ITEM2["DATE_RANGE"] = "date_range";
  FORM_ITEM2["DATETIME_RANGE"] = "datetime_range";
  FORM_ITEM2["TIME_RANGE"] = "time_range";
  return FORM_ITEM2;
})(FORM_ITEM || {});
export var TABLE_ITEM_KEY = /* @__PURE__ */ ((TABLE_ITEM_KEY2) => {
  TABLE_ITEM_KEY2["ID"] = "id";
  TABLE_ITEM_KEY2["SORT"] = "key_sort";
  TABLE_ITEM_KEY2["ACTION"] = "key_action";
  TABLE_ITEM_KEY2["DELETE"] = "key_delete";
  TABLE_ITEM_KEY2["EDIT"] = "key_edit";
  TABLE_ITEM_KEY2["LINE"] = "key_line";
  TABLE_ITEM_KEY2["HEADLINE"] = "key_headline";
  TABLE_ITEM_KEY2["BLANK_COLUMN"] = "key_blank_column";
  TABLE_ITEM_KEY2["BLANK"] = "key_blank";
  TABLE_ITEM_KEY2["COMMENT"] = "key_comment";
  TABLE_ITEM_KEY2["NOTE"] = "key_note";
  TABLE_ITEM_KEY2["ATTENTION"] = "key_attention";
  TABLE_ITEM_KEY2["WARNING"] = "key_warning";
  TABLE_ITEM_KEY2["MODAL_LINK"] = "key_modallink";
  TABLE_ITEM_KEY2["ICON_OR_LINK"] = "key_iconOrlink";
  TABLE_ITEM_KEY2["OSS_THUMBNAIL"] = "key_oss_image_url";
  return TABLE_ITEM_KEY2;
})(TABLE_ITEM_KEY || {});
export const CONST_TABLE_ITEM = {
  ID: {
    key: "id" /* ID */,
    type: "input" /* INPUT */,
    title: "ID",
    align: "center",
    width: 160,
    strWidth: "160px",
    minWidth: "160px",
    maxWidth: "160px",
    readOnly: true,
    fixed: true,
    dialog: true
  },
  OSS_THUMBNAIL: {
    key: "key_oss_image_url" /* OSS_THUMBNAIL */,
    type: "input" /* INPUT */,
    title: "\u30B5\u30E0\u30CD\u30A4\u30EB",
    align: "center",
    width: 80,
    strWidth: "80px",
    minWidth: "80px",
    maxWidth: "80px",
    listOnly: true,
    fixed: true
  },
  BLANK: {
    key: "key_blank" /* BLANK */,
    type: "key_blank" /* BLANK */,
    title: "",
    sortable: false,
    align: "center",
    maxWidth: "100%"
  },
  ACTION: {
    key: "key_action" /* ACTION */,
    type: "key_action" /* ACTION */,
    title: "ACTION",
    sortable: false,
    align: "center",
    width: 40,
    strWidth: "40px",
    minWidth: "40px",
    maxWidth: "40px"
  },
  DELETE: {
    key: "key_delete" /* DELETE */,
    type: "key_delete" /* DELETE */,
    title: "\u524A\u9664",
    sortable: false,
    align: "center",
    width: 40,
    strWidth: "40px",
    minWidth: "40px",
    maxWidth: "40px"
  },
  LINE: {
    key: "key_line" /* LINE */,
    type: "key_line" /* LINE */,
    title: "line",
    fixed: false
  },
  HEADLINE: {
    key: "key_headline" /* HEADLINE */,
    type: "headline" /* HEADLINE */,
    title: "HEADLINE"
  },
  BLANK_COLUMN: {
    key: "key_blank_column" /* BLANK_COLUMN */,
    type: "key_blank_column" /* BLANK_COLUMN */,
    title: "BLANK_COLUMN"
  },
  COMMENT: {
    key: "key_comment" /* COMMENT */,
    type: "key_comment" /* COMMENT */,
    title: "line"
  },
  NOTE: {
    key: "key_note" /* NOTE */,
    type: "key_note" /* NOTE */,
    title: "line"
  },
  ATTENTION: {
    key: "key_attention" /* ATTENTION */,
    type: "key_attention" /* ATTENTION */,
    title: "line"
  },
  WARNING: {
    key: "key_warning" /* WARNING */,
    type: "key_warning" /* WARNING */,
    title: "line"
  }
};
export const defaultBase = {
  label: {
    size: "14px",
    width: 100,
    weight: "normal"
  },
  size: "14px",
  mode: "inline" /* INLINE */,
  edit: true,
  disable: false
};
export var SELECT_INPUT_TYPE = /* @__PURE__ */ ((SELECT_INPUT_TYPE2) => {
  SELECT_INPUT_TYPE2["LINK"] = "link";
  SELECT_INPUT_TYPE2["TEXT"] = "text";
  SELECT_INPUT_TYPE2["BUTTON"] = "button";
  SELECT_INPUT_TYPE2["INPUT"] = "input";
  SELECT_INPUT_TYPE2["FRAME"] = "frame";
  SELECT_INPUT_TYPE2["STRETCH_ICON"] = "stretch_icon";
  return SELECT_INPUT_TYPE2;
})(SELECT_INPUT_TYPE || {});
export var SELECT_LIST_TYPE = /* @__PURE__ */ ((SELECT_LIST_TYPE2) => {
  SELECT_LIST_TYPE2["RADIO"] = "radio";
  SELECT_LIST_TYPE2["CHECK"] = "check";
  SELECT_LIST_TYPE2["DEFAULT"] = "default";
  return SELECT_LIST_TYPE2;
})(SELECT_LIST_TYPE || {});
export var HOVER_TYPE = /* @__PURE__ */ ((HOVER_TYPE2) => {
  HOVER_TYPE2["DARKEN"] = "darken";
  HOVER_TYPE2["LIGHTEN"] = "lighten";
  return HOVER_TYPE2;
})(HOVER_TYPE || {});
export var TOGGLE_TYPE = /* @__PURE__ */ ((TOGGLE_TYPE2) => {
  TOGGLE_TYPE2["RADIO"] = "radio";
  TOGGLE_TYPE2["CHECK"] = "check";
  TOGGLE_TYPE2["SQUARE"] = "square";
  TOGGLE_TYPE2["CIRCLE"] = "circle";
  TOGGLE_TYPE2["OTHER"] = "other";
  return TOGGLE_TYPE2;
})(TOGGLE_TYPE || {});
export var CALENDAR_TYPE = /* @__PURE__ */ ((CALENDAR_TYPE2) => {
  CALENDAR_TYPE2["DISPLAY"] = "display";
  CALENDAR_TYPE2["DAY_INPUT"] = "dayInput";
  CALENDAR_TYPE2["DAY_CALENDAR"] = "dayCalendar";
  CALENDAR_TYPE2["PERIOD_INPUT"] = "periodInput";
  CALENDAR_TYPE2["PERIOD_CALENDAR"] = "periodCalendar";
  CALENDAR_TYPE2["MULTI_DAYS_INPUT"] = "multiDaysInput";
  CALENDAR_TYPE2["MULTI_DAYS_CALENDAR"] = "multiDaysCalendar";
  CALENDAR_TYPE2["PERMITTED_INPUT"] = "permittedInput";
  CALENDAR_TYPE2["PERMITTED_CALENDAR"] = "permittedCalendar";
  return CALENDAR_TYPE2;
})(CALENDAR_TYPE || {});
export var DATE_FORMAT = /* @__PURE__ */ ((DATE_FORMAT2) => {
  DATE_FORMAT2["default"] = "YYYY-MM-DD";
  DATE_FORMAT2["yymm"] = "YYYY-MM";
  DATE_FORMAT2["yymm_jp"] = "\u5E74\u6708";
  DATE_FORMAT2["yymmdd_jp"] = "\u5E74\u6708\u65E5";
  return DATE_FORMAT2;
})(DATE_FORMAT || {});
export var CLOCK_TYPE = /* @__PURE__ */ ((CLOCK_TYPE2) => {
  CLOCK_TYPE2["DISPLAY"] = "display";
  CLOCK_TYPE2["INPUT_DIAL"] = "input_dial";
  CLOCK_TYPE2["DIAL"] = "dial";
  return CLOCK_TYPE2;
})(CLOCK_TYPE || {});
export var DROP_TYPE = /* @__PURE__ */ ((DROP_TYPE2) => {
  DROP_TYPE2["DROP_AREA"] = "area";
  DROP_TYPE2["DIAPLAY_FILES"] = "files";
  DROP_TYPE2["DISPLAY_FILE"] = "file";
  DROP_TYPE2["FORM"] = "form";
  return DROP_TYPE2;
})(DROP_TYPE || {});
export var FILE_UPLOAD_TYPE = /* @__PURE__ */ ((FILE_UPLOAD_TYPE2) => {
  FILE_UPLOAD_TYPE2["SINGLE"] = "single";
  FILE_UPLOAD_TYPE2["MULTI"] = "multi";
  return FILE_UPLOAD_TYPE2;
})(FILE_UPLOAD_TYPE || {});
export var FILE_EXTENSION = /* @__PURE__ */ ((FILE_EXTENSION2) => {
  FILE_EXTENSION2["TEXT_TXT"] = "txt";
  FILE_EXTENSION2["TEXT_CSV"] = "csv";
  FILE_EXTENSION2["TEXT_XLS"] = "xls";
  FILE_EXTENSION2["TEXT_XLSX"] = "xlsx";
  FILE_EXTENSION2["AUDIO_MP3"] = "mp3";
  FILE_EXTENSION2["MOVIE_MP4"] = "mp4";
  FILE_EXTENSION2["IMG_PNG"] = "png";
  FILE_EXTENSION2["IMG_JPG"] = "jpg";
  FILE_EXTENSION2["IMG_JPEG"] = "jpeg";
  FILE_EXTENSION2["IMG_GIF"] = "gif";
  FILE_EXTENSION2["ZIP"] = "zip";
  FILE_EXTENSION2["APP_JSON"] = "json";
  FILE_EXTENSION2["APP_PDF"] = "pdf";
  return FILE_EXTENSION2;
})(FILE_EXTENSION || {});
export var FILE_TYPE = /* @__PURE__ */ ((FILE_TYPE2) => {
  FILE_TYPE2["TEXT_TXT"] = "text/plain";
  FILE_TYPE2["TEXT_CSV"] = "text/csv";
  FILE_TYPE2["TEXT_XLS"] = "application/vnd.ms-excel";
  FILE_TYPE2["TEXT_XLSX"] = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet";
  FILE_TYPE2["AUDIO_MP3"] = "audio/mpeg";
  FILE_TYPE2["MOVIE_MP4"] = "video/mp4";
  FILE_TYPE2["IMG_PNG"] = "image/png";
  FILE_TYPE2["IMG_JPG"] = "image/jpeg";
  FILE_TYPE2["IMG_JPEG"] = "image/jpeg";
  FILE_TYPE2["IMG_GIF"] = "image/gif";
  FILE_TYPE2["MAC_ZIP"] = "application/zip";
  FILE_TYPE2["WIN_ZIP"] = "application/x-zip-compressed";
  FILE_TYPE2["APP_JSON"] = "application/json";
  FILE_TYPE2["APP_PDF"] = "application/pdf";
  return FILE_TYPE2;
})(FILE_TYPE || {});
